src/core_modules/capture-core/components/DataEntries/SingleEventRegistrationEntry/SingleEventRegistrationEntry.container.js

```diff
@@ -13,13 +13,14 @@ import { defaultDialogProps as dialogConfig } from '../../Dialogs/DiscardDialog.
 import { getOpenDataEntryActions } from './DataEntryWrapper/DataEntry';
 import type { ContainerProps, StateProps, MapStateToProps } from './SingleEventRegistrationEntry.types';
 import { useCategoryCombinations } from '../../DataEntryDhis2Helpers/AOC/useCategoryCombinations';
+import { itemId } from './DataEntryWrapper/DataEntry/helpers/constants';
 
 const inEffect = (state: ReduxState) => dataEntryHasChanges(state, 'singleEvent-newEvent') || state.newEventPage.showAddRelationship;
 
 const makeMapStateToProps = (): MapStateToProps => {
     const eventAccessSelector = makeEventAccessSelector();
     return (state: ReduxState, { id }: ContainerProps): StateProps => ({
-        ready: state.dataEntries[id],
+        ready: state.dataEntries[id]?.itemId === itemId,
```

Comment:
From a code review perspective it's, to be honest, still not quite clear how this change addresses the described issue... Would the decent thing for me to do be to put in some research effort and figure it out myself, or is it reasonable to ask for further explanation? @JoakimSM 

I don't doubt it works when put to the test, but I'm not sure I should give my approval to the code if I don't understand why it works.

Reply:
I see, it looks like all `dataEntries`-prefixed elements in the redux store are initialised simultaneously. So when `dataEntries.singleEvent.itemId'` changes to `newEvent`, we know that `dataEntriesFieldsUI.singleEvent-newEvent` must also have been initialised :+1: